ttnn/cpp/ttnn/tensor/tensor.hpp

```diff
@@ -139,6 +139,25 @@ struct Tensor {
 
     std::vector<Device*> get_workers(bool blocking = false) const;
 
+    // Converts a buffer of elements of type `T` to a `Tensor`.
+    // Elements are assumed to be stored in row-major order. The size of the span and the type have to match `spec`.
+    //
+    // TODO: tilized layouts and reduced precision types are currently not supported.
+    template <typename T>
+    static Tensor from_span(tt::stl::Span<const T> buffer, const TensorSpec& spec);
+
+    // Same as `from_span`, but takes a vector instead.
+    template <typename T>
+    static Tensor from_vector(const std::vector<T>& buffer, const TensorSpec& spec) {
+        return from_span(tt::stl::Span<const T>(buffer.data(), buffer.size()), spec);
+    }
+
+    // Converts a `Tensor` to a buffer of elements of type `T`.
+    // Elements in the buffer will be stored in row-major order. The type of the elements has to match that of the
+    // `Tensor`.
+    template <typename T>
+    std::vector<T> to_vector() const;
```

Comment:
I assume this is only currently supported for a subset of cases. Again, afaik tile layout is not handled, block float formats not supported, sharding, including logical sharding is not supported. Right?

Reply:
Added a clarifying comment. Does tilized `std::vector` make sense though? For bf4/bf8, my plan was to just convert them into float and return a row-major slice.

Reply:
thanks!

yes, tilized output does not makes sense. returning floats is right.